internal/cephfs/volume.go

```diff
@@ -102,23 +102,26 @@ func (vo *volumeOptions) getSubVolumeInfo(ctx context.Context, volID volumeID) (
 		Path:     info.Path,
 		Features: make([]string, len(info.Features)),
 	}
-	bc, ok := info.BytesQuota.(fsAdmin.ByteCount)
-	if !ok {
-		// If info.BytesQuota == Infinite (in case it is not set)
-		// or nil (in case the subvolume is in snapshot-retained state),
-		// just continue without returning quota information.
-		// TODO: make use of subvolume "state" attribute once
-		// https://github.com/ceph/go-ceph/issues/453 is fixed.
-		if !(info.BytesQuota == fsAdmin.Infinite || info.BytesQuota == nil) {
-			return nil, fmt.Errorf("subvolume %s has unsupported quota: %v", string(volID), info.BytesQuota)
+
+	// SubVolumeInfo returns all the attributes only when subvolume
+	// state is Complete.
+	if info.State == fsAdmin.StateComplete {
+		bc, ok := info.BytesQuota.(fsAdmin.ByteCount)
+		if !ok {
+			// If info.BytesQuota == Infinite (in case it is not set)
+			// just continue without returning quota information.
+			if info.BytesQuota != fsAdmin.Infinite {
```

Comment:
@agarwal-mudit for few ceph versions the state wont be available in subvolume info right?

```bash
ceph fs subvolume info  myfs csi-vol-0b5b6e7f-6f85-11eb-bc33-0242ac110007 csi
{
    "atime": "2021-02-15 11:58:07",
    "bytes_pcent": "0.00",
    "bytes_quota": 1073741824,
    "bytes_used": 0,
    "created_at": "2021-02-15 11:58:07",
    "ctime": "2021-02-15 11:58:07",
    "data_pool": "myfs-data0",
    "features": [
        "snapshot-clone",
        "snapshot-autoprotect"
    ],
    "gid": 0,
    "mode": 16895,
    "mon_addrs": [
        "192.168.39.222:6789"
    ],
    "mtime": "2021-02-15 11:58:07",
    "path": "/volumes/csi/csi-vol-0b5b6e7f-6f85-11eb-bc33-0242ac110007/689a1127-dd52-41e0-ac58-48ba8afeda0e",
    "pool_namespace": "",
    "type": "subvolume",
    "uid": 0
}
```

Reply:
Yes, that's why I have kept this a Draft. I wanted suggestion. This field will be there for versions which have snapshot retention feature.

Reply:
If we want to fix it we need to check the snapshot-retention feature if it's present check the completed state

```go
if info.features.Contains("snapshot-retention") && info.State == fsAdmin.StateComplete {
// handle size and infinite
}else{
// handle size,infinite and nil
}
```
@agarwal-mudit do you see any advantage with this PR?

Reply:
have removed this